Formula/git.rb

```diff
@@ -76,9 +76,37 @@ def install
       args += %w[NO_OPENSSL=1 APPLE_COMMON_CRYPTO=1]
     end
 
+    git_core = libexec/"git-core"
+    # By setting TKFRAMEWORK to a non-existent directory we ensure that
+    # the git makefiles don't install a .app for git-gui
+    # We also tell git to use a wrapper script for wish instead of the
+    # broken system-provided wish.
+    # See https://github.com/Homebrew/homebrew-core/issues/36390
+    args += %W[
+      TKFRAMEWORK="/we/do/not/want/to/install/a/.app/for/git-gui"
+      TCLTK_PATH=#{git_core/"git-wish-wrapper"}
+    ]
+
     system "make", "install", *args
 
-    git_core = libexec/"git-core"
+    # Create the wrapper for Wish that prefers the homebrew version of tlc-tk
+    (git_core/"git-wish-wrapper").write <<~EOS
+      #!/bin/sh
```

Comment:
This is a lot of code to include in the formula. I'd suggest something similar be sent to Git upstream instead (or a `configure` check to ensure a certain version of Tcl/tk is used).